src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1595,7 +1597,7 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
-        public static ProtocolMessages.Module ConvertToModule(in MODULE_INFO debugModuleInfo, IDebugModule2 module, bool isLoad)
+        private ProtocolMessages.Module ConvertToModule(in MODULE_INFO debugModuleInfo, IDebugModule2 module, bool isLoad)
```

Comment:
I would convert this method to:
```suggestion
        private ProtocolMessages.Module ConvertToModule(in MODULE_INFO debugModuleInfo, int moduleId)
```

and in `HandleIDebugModuleLoadEvent2`
```
int? moduleId = null;
ModuleEvent.ReasonValue reason = ModuleEvent.ReasonValue.Unknown;
if (isLoad)
{
   moduleId = RegisterDebugModule(module);
   reason = ModuleEvent.ReasonValue.New;
}
else
{
    moduleId = ReleaseDebugModule(module);
    reason = ModuleEvent.ReasonValue.Removed;
}

if (moduleId != null)
{
   mod = ConvertToModule(debugModuleInfos[0], moduleId);
   Protocol.SendEvent(new ModuleEvent(reason, mod));
}
```

Reply:
I will move the IsUserCode assignment to HandleIDebugModuleLoadEvent2().

Reply:
You could modify this to take a IDebugModule2 and just call IDebugModule2.GetInfo in here.

Reply:
I am confused on how to handle HandleModulesRequestAsync().

Reply:
Any modules enumerated from `IEnumDebugModules2.EnumModules` would be new modules. Treat them like a load event.